### PR TITLE
fix bug for TableFactoryPrecompiled

### PR DESCRIPTION
**Pull request by qwdarrenyin**

### CODE REVIEW

File: libprecompiled/TableFactoryPrecompiled.cpp [hunk 1]
```diff
@@ -53,6 +53,46 @@ std::string TableFactoryPrecompiled::toString()
     return "TableFactory";
 }
 
+
+void TableFactoryPrecompiled::checkFieldNameHasUnderline(
+    const std::string& fieldName, const std::string& tableName)
```
Comment (bxq2011hust) [inline libprecompiled/TableFactoryPrecompiled.cpp]: why you need tableName？

File: libprecompiled/TableFactoryPrecompiled.cpp [hunk 2]
```diff
@@ -53,6 +53,46 @@ std::string TableFactoryPrecompiled::toString()
     return "TableFactory";
 }
 
+
+void TableFactoryPrecompiled::checkFieldNameHasUnderline(
```
Comment (bxq2011hust) [inline libprecompiled/TableFactoryPrecompiled.cpp]: I think this function is unnecessary

File: libprecompiled/TableFactoryPrecompiled.cpp [hunk 3]
```diff
@@ -53,6 +53,46 @@ std::string TableFactoryPrecompiled::toString()
...
+    if (fieldName.size() == 0 || fieldName[0] == '_')
+    {
+        STORAGE_LOG(ERROR) << LOG_DESC("error key") << LOG_KV("field name", fieldName)
+                           << LOG_KV("table name", tableName);
+        BOOST_THROW_EXCEPTION(StorageException(
+            CODE_TABLE_INVALIDATE_FIELD, std::string("invalidate field:") + fieldName));
+    }
+}
+
+void TableFactoryPrecompiled::checkFieldNameValidate(
```
Comment (bxq2011hust) [inline libprecompiled/TableFactoryPrecompiled.cpp]: I recommend keeping the same parameters' order of createTable

File: libprecompiled/TableFactoryPrecompiled.h [hunk 4]
```diff
@@ -68,6 +68,11 @@ class TableFactoryPrecompiled : public Precompiled
 
 private:
     std::shared_ptr<dev::storage::TableFactory> m_memoryTableFactory;
+
+private:
```
Comment (bxq2011hust) [inline libprecompiled/TableFactoryPrecompiled.h]: duplicated private

File: libstorage/Common.h [hunk 5]
```diff
@@ -84,6 +84,8 @@ const int CODE_TABLE_FILED_LENGTH_OVERFLOW = -50003;
 const int CODE_TABLE_FILED_TOTALLENGTH_OVERFLOW = -50004;
 const int CODE_TABLE_KEYVALUE_LENGTH_OVERFLOW = -50005;
 const int CODE_TABLE_FIELDVALUE_LENGTH_OVERFLOW = -50006;
+const int CODE_TABLE_DUMPLICATE_FIELD = -50007;
```
Comment (cyjseagull) [inline libstorage/Common.h]: The `console` or `web3sdk` should support the error code in the future version.

**Pull request merged**